### ISSUE TITLE
wiki page edit notifications

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#396]
for when someone edits a page. 

This'd only really be useful on pages that shouldn't/don't get edited often. And it'd include (potentially) the changes made and the username of the editor. These could have the same "Look like spam?" links the research note notifications have.

**Comment by StlMaris123** [comment]
Hello @jywarren, could this issue be broken down a little bit. looks a little bulky to me.

**Comment by jywarren** [comment]
I agree. Do you want to try to make a checklist in the comments here, and
propose a way to break it up? I think perhaps just getting an experimental
ActiveJob (ActionJob?) task running and passing all tests would be a good
first step. Then one which sends an email regularly (although this could be
annoying, but we could create a Gmail rule to filter it out) -- that way we
can confirm that emails are being sent at all on a schedule.

Then we could move on to more advanced integration.

On Thu, May 11, 2017 at 2:53 PM, StellaMaris Njage <notifications@github.com

**Comment by StlMaris123** [comment]
Yes. I'll send the breakdown tomorrow morning.

**Comment by StlMaris123** [comment]
We could divide them into two tasks:
1. **No edits for over an hour**
- Acceptance tests
- Send only one email if no edits have occurred within an hour.

2. **Several subsequent edits**
-  Acceptance tests
- Send only one email for all subsequent edits

However, I just thought we could have one email once a wiki page is created. We could have two links:
- edit link
- submit final

One can keep editing the wiki page until finally contented the click `submit final`. That is is when guys get notified about the wiki updates. That way we don't have to worry about the number of edits performed. What do you think?

**Comment by ujithaperera** [comment]
Hi @StlMaris123,

BTW I am Ujitha Perera. I also agree with the @jywarren . I feel that first we should make sure that active Job is up and running. Since this is a new module to the code base, have to create new active Jobs without affecting to the current functionalities. 

And since we are using rails 3.2 active jobs don't support by default. hence we have to add active jobs as a gem to the codebase and then the queuing process. Since our DB size is not small definitely we may have to go with parallel process. Perhaps we may able to run email batches within the rails app. But it would be a huge load for the rails app and other function might not work within that email processing time. Hope @jywarren will explain the actual requirement. 

@StlMaris123 are you familiar with active jobs and background jobs ? if not, installing Redis and scheduling background jobs would be a great hands on experience for you to begin with.

**Comment by StlMaris123** [comment]
Hello Ujitha,
 I am not familiar but i will have a look at them.

**Comment by StlMaris123** [comment]
@jywarren Please help me in breaking down the tasks into more distinct steps.

**Comment by jywarren** [comment]
Hi, @StlMaris123 - i'll reply to this soon, sorry i've been tied up!

**Comment by jywarren** [comment]
Hi, @StlMaris123  -- sorry for the delay but I've posted a proposed 2-phase plan to attempt the above as well as the next step. Check out #1421 and tell me what you think!

**Comment by SidharthBansal** [comment]
If @jywarren you think this is a minor project OR can be broken into 2-3 minor projects then we can include this as hall of fame task in GCI
